pkg/apis/operator/v1alpha1/validation/validation.go

```diff
@@ -210,6 +220,31 @@ func validateGardenerAPIServerConfig(config *operatorv1alpha1.GardenerAPIServerC
 		}
 	}
 
+	if config.EncryptionConfig != nil {
+		seenResources := sets.New[string]()
+
+		for i, resource := range config.EncryptionConfig.Resources {
+			idxPath := fldPath.Child("encryptionConfig", "resources").Index(i)
+			if seenResources.Has(resource) {
+				allErrs = append(allErrs, field.Duplicate(idxPath, resource))
+			}
+
+			if gardenerutils.DefaultGardenerResourcesForEncryption().Has(resource) {
+				allErrs = append(allErrs, field.Forbidden(idxPath, fmt.Sprintf("%q are always encrypted", resource)))
+			}
+
+			if !gardenerutils.IsServedByGardenerAPIServer(resource) {
+				allErrs = append(allErrs, field.Invalid(idxPath, resource, "should be a resource served by gardener-apiserver. ie; should have any of the suffixes {core,operations,settings,seedmanagement}.gardener.cloud"))
+			}
+
+			if strings.HasPrefix(resource, "*.") {
```

Comment:
A resource `*apps` won't work anyways, right?

Reply:
I guess there is no disadvantage using `*` (or is there?) because if it will once work (maybe `*` for all resources), we don't have to adjust the code.

Reply:
All resources is `*.*`, but there is no disadvantage, I can change it.